syntax = "proto3";

package torrentparser;
option go_package = "github.com/ipromknight/zilean/src/GoPttServer/torrentparser";
option csharp_namespace = "Zilean.Proto.TorrentParser";

service TorrentParser {
  rpc ParseTitles (stream TorrentTitleRequest) returns (stream TorrentTitleResponse);
  rpc Shutdown (ShutdownRequest) returns (ShutdownResponse);
}

message TorrentTitleRequest {
  string info_hash = 1;
  string title = 2;
}

message TorrentTitleResponse {
  string info_hash = 1;
  string original_title = 2;
  repeated string audio = 3;
  string bit_depth = 4;
  repeated string channels = 5;
  string codec = 6;
  bool commentary = 7;
  bool complete = 8;
  string container = 9;
  bool convert = 10;
  string date = 11;
  bool documentary = 12;
  bool dubbed = 13;
  string edition = 14;
  string episode_code = 15;
  repeated int32 episodes = 16;
  bool extended = 17;
  string extension = 18;
  string group = 19;
  repeated string hdr = 20;
  bool hardcoded = 21;
  repeated string languages = 22;
  string network = 23;
  bool proper = 24;
  string quality = 25;
  string region = 26;
  bool remastered = 27;
  bool repack = 28;
  string resolution = 29;
  bool retail = 30;
  repeated int32 seasons = 31;
  string site = 32;
  string size = 33;
  bool subbed = 34;
  string three_d = 35;
  string title = 36;
  bool uncensored = 37;
  bool unrated = 38;
  bool upscaled = 39;
  repeated int32 volumes = 40;
  string year = 41;
}

message ShutdownRequest {}
message ShutdownResponse {}